src/DocumentationViewer.js

```diff
@@ -110,117 +108,104 @@ export default function DocumentationViewer({ application }) {
     }
   });
 
-  let promise_load;
-  const filter_docs = [
-    "atk",
-    "javascriptcoregtk-4.1",
-    "libhandy-1",
-    "libnotify-0",
-    "webkit2gtk-4.1",
-    "webkit2gtk-web-extension-4.1",
-  ];
+  const search_model = builder.get_object("search_model");
+  const sorter = builder.get_object("search_sorter");
+  sorter.expression = expr;
+  search_model.connect("selection-changed", () => {
+    const uri = search_model.selected_item.uri;
+    webview.load_uri(uri);
+  });
+
   async function open() {
-    if (!promise_load)
-      promise_load = getDirs(base_path, filter_docs)
-        .then((dirs) => createIndex(base_path, dirs))
-        .then((indexes) => {
-          browse_list_view.model = createBrowseListModel(
-            base_path,
-            indexes,
-            webview,
-          );
-          search_list_view.model = createSearchListModel(
-            base_path,
-            indexes,
-            webview,
-            filter,
-          );
-        });
-    await promise_load;
-    window.present();
+    const root_model = Gio.ListStore.new(DocumentationPage);
+
+    scanLibraries(root_model, base_path)
+      .then(() => {
+        browse_list_view.model.selected = 12;
+        const search_model = flattenModel(root_model);
+        filter_model.model = search_model;
+      })
+      .catch(console.error);
+
+    browse_list_view.model = createBrowseSelectionModel(root_model, webview);
```

Comment:
```suggestion
    try {
      const root_model = Gio.ListStore.new(DocumentationPage);

      scanLibraries(root_model, base_path)
        .then(() => {
          browse_list_view.model.selected = 12;
          const search_model = flattenModel(root_model);
          filter_model.model = search_model;
        })
        .catch(console.error);

      browse_list_view.model = createBrowseSelectionModel(root_model, webview);
    } catch(error) {
      console.error(error);
    }
```

I don't know where it makes most sense to `catch`, but there should either be a `try..catch` here or a `catch()` when it's invoked.